subiquity/server/controllers/tests/test_cmdlist.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from unittest import IsolatedAsyncioTestCase, mock
+
+from subiquity.server.controllers import cmdlist
+from subiquity.server.controllers.cmdlist import (
+    Command,
+    EarlyController,
+    LateController,
+)
+from subiquitycore.tests.mocks import make_app
+
+
+class TestCmdListController(IsolatedAsyncioTestCase):
+    def setUp(self):
+        self.mocked_arun = mock.patch.object(cmdlist, "arun_command")
+        snap_env_vars = {
+            "LD_LIBRARY_PATH": "/var/lib/snapd/lib/gl",
+        }
+        self.mocked_os_environ = mock.patch.dict("os.environ", snap_env_vars)
+
+
+class TestEarlyController(TestCmdListController):
+    def setUp(self):
+        super().setUp()
+        self.controller = EarlyController(make_app())
+        self.controller.cmds = [Command(args="some-command", check=False)]
+
+    async def test_no_library_path_on_run(self):
+        with (
+            self.mocked_arun as faked_process_call,
```

Comment:
python3.8 is being annoying (hopefully we will get rid of it soon !) but you can work around the issue by dropping the parentheses :+1: